lib/std/os/linux.zig

```diff
@@ -440,28 +440,51 @@ pub fn read(fd: i32, buf: [*]u8, count: usize) usize {
 }
 
 pub fn preadv(fd: i32, iov: [*]const iovec, count: usize, offset: i64) usize {
-    const offset_halves = splitValueLE64(offset);
-    return syscall5(
-        .preadv,
-        @bitCast(usize, @as(isize, fd)),
-        @ptrToInt(iov),
-        count,
-        offset_halves[0],
-        offset_halves[1],
-    );
+    if (usize_bits < 64) {
```

Comment:
It would better match the kernel logic if instead of branching here we just passed `@truncate(usize, @bitCast(u64, offset))` and `@truncate(usize, @bitCast(u64, offset) >> 32)`.

Reply:
Also it would be very nice to link the linux commit I found with the explanation in the source code to prevent anyone being further confused by this in the future.

Reply:
@ifreund Thanks for the comment.
For the high end, do you think this would be better?
`if (usize_bits < 64) @truncate(usize, @bitCast(u64, offset) >> usize_bits) else 0`

I think this would cause less confusion.

Reply:
Yeah, that seems nice to me. A good comment would also go a long way to prevent confusion.

Reply:
Agree. I just uploaded a new version.